erpnext/public/js/controllers/transaction.js

```diff
@@ -1911,7 +1911,7 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 	set_query_for_item_tax_template: function(doc, cdt, cdn) {
 		var item = frappe.get_doc(cdt, cdn);
 		if(!item.item_code) {
-			frappe.throw(__("Please enter Item Code to get item taxes"));
+			// Do nothing
 		} else {
```

Comment:
In the case of not having an item code for manual entries, `set_query_for_item_tax_template` is only triggered if a user triggers the field.
The **Item** Tax Template is tied to an **Item**. And so if there is no item code, just an item name for quick invoicing, it should rightfully throw this error.
In majority of the cases Item codes are used, for which this validation is important.

I'd question why the user is trying to select an Item Tax Template without a valid item ? Don't see anything to fix here. Just my views.

Reply:
> In the case of not having an item code for manual entries, set_query_for_item_tax_template is only triggered if a user triggers the field.

For clarity: if the user triggers the `item_tax_template` field

> The Item Tax Template is tied to an Item. And so if there is no item code, just an item name for quick invoicing, it should rightfully throw this error.

I'm afraid you're incorrect. The Item Tax Template is not tied to an item, the template content is saved in the `item_tax_rate` field on the `* Invoice Item` record as json. There is no reason for there to be an `item_code` at all. This storage is vitally important so that Invoices are fixed when submitted, otherwise changes to the template will affect past records.

I have tested this change and it works well, but if left as it is the line prevents arbitrary line items on a Sales Order. I recommend you try the standard behaviour.

> In majority of the cases Item codes are used, for which this validation is important.

But not in *all* cases. There are numerous perfectly valid use-cases that should *never* require an item_code:

- Purchase invoices for one-off items, like lunch bought by the company
- Sales of bespoke items